### PR TITLE
feat(pie-docs): DSW-2345 enhance contentLayout

### PR DESCRIPTION
**Pull request by leksaBoiko**
## Describe your changes (can list changeset entries if preferable)

Feature url: https://pr1767-docs.pie.design/components/textarea/#auto-resize

[Updated] - Textarea overview page layout for "Behaviours" section
[Updated] - the `contentLayout` container to display the content of the specified number of columns

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.

### Reviewer 1 - @xander-marjoram 
- [x] I have reviewed the `PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 - @jamieomaguire 
- [x] I have reviewed the `PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by xander-marjoram** [comment]
Visually this looks great, and it's nicely done.

In terms of the code, I think it could be improved to be more flexible. A three-column layout should be the default for the larger viewports and I don't think we need a seperate class or a `columns` property to control this.

The card list on the docs homepage (For designers, For engineers, Our libraries) I think is quite a good example of how I'd prefer something like this to behave.

**Comment by leksaBoiko** [comment]
> Visually this looks great, nicely done.
> 
> In terms of the code, I think it could be improved to be more flexible. A three-column layout should be the default for the larger viewports and I don't think we need a seperate class or a `columns` property to control this.
> 
> The card list on the docs homepage (For designers, For engineers, Our libraries) I think is quite a good example of how I'd prefer something like this to behave.

Based on feedback from the designers, it seems we need to include a separate 3-column layout, as there might be instances where both 2-column and 3-column layouts are displayed on the same screen resolution. However, it would be a good idea to confirm this with the designers for clarity.

**Comment by tingting-lau** [comment]
Hey leksa is right, we use both 2-column and 3-column layout in the documentation

**Review by xander-marjoram — APPROVED** [review]

File: apps/pie-docs/src/assets/styles/components/_contentLayout.scss [hunk 1]
```diff
@@ -13,6 +13,18 @@
...
+    @include f.media('>=huge') {
+      flex-flow: row wrap;
+
+      > * {
+        // stylelint-disable-next-line declaration-block-no-redundant-longhand-properties
+        flex: 0 1 calc(33.333% - var(--dt-spacing-e)); // Three equal-width columns
+      }
+    }
+  }
 }
```
Comment (jamieomaguire) [inline apps/pie-docs/src/assets/styles/components/_contentLayout.scss:28]: I don't think this is ideal. We should be writing the CSS more fluidly so it renders 3 items on a single row. For example in a grid: `grid-template-rows: repeat(1fr, 3);`.

Why do we specifically need to write separate code for 3 items? 

If I have 2 or 3, but tell the container to display then on a row, why is it different for 3?

Furthermore, what if we want to display 4 or 5 in future?

**Review by jamieomaguire — APPROVED** [review]

**Pull request merged**